tests/test_tree_stats.py

```diff
@@ -889,56 +934,255 @@ def test_case_2(self):
+                               branch_true_diversity_01)
+        self.assertAlmostEqual(py_branch_tsc.tree_stat(A, f),
+                               branch_true_diversity_01)
 
         # mean divergence between 0, 1 and 0, 2
         A = [[0, 1], [0, 2]]
         n = [len(a) for a in A]
 
         def f(x):
-            return float(x[0]*(n[1]-x[1]) + (n[0]-x[0])*x[1])/4.0
+            return float(x[0]*(n[1]-x[1]) + (n[0]-x[0])*x[1])/8.0
 
-        # branch lengths:
-        self.assertAlmostEqual(branch_length_diversity(ts, A[0], A[1]),
-                               true_mean_diversity)
-        self.assertAlmostEqual(tsc.tree_stat(A, tupleize(f)),
-                               true_mean_diversity)
-        self.assertAlmostEqual(tree_stat_node_iter(ts, A, f),
-                               true_mean_diversity)
+        # tree lengths:
+        self.assertAlmostEqual(py_branch_tsc.tree_length_diversity(A[0], A[1]),
+                               branch_true_mean_diversity)
+        self.assertAlmostEqual(branch_tsc.tree_stat(A, f),
+                               branch_true_mean_diversity)
+        self.assertAlmostEqual(py_branch_tsc.tree_stat(A, f),
+                               branch_true_mean_diversity)
 
         # Y-statistic for (0/12)
         A = [[0], [1, 2]]
 
         def f(x):
-            return ((x[0] == 1) and (x[1] == 0)) or ((x[0] == 0) and (x[1] == 2))
+            return float(((x[0] == 1) and (x[1] == 0))
+                         or ((x[0] == 0) and (x[1] == 2)))/2.0
+
+        # tree lengths:
+        self.assertAlmostEqual(py_branch_tsc.Y3([0], [1], [2]), branch_true_Y)
+        self.assertAlmostEqual(branch_tsc.tree_stat(A, f), branch_true_Y)
+        self.assertAlmostEqual(py_branch_tsc.tree_stat(A, f), branch_true_Y)
+
+        # sites:
+        site_tsc_Y = site_tsc.Y3([[0], [1], [2]], [0.0, 1.0])[0][0]
+        py_site_tsc_Y = py_site_tsc.Y3([0], [1], [2], 0.0, 1.0)
+        print(site_tsc_Y, py_site_tsc_Y, site_true_Y)
+        self.assertAlmostEqual(site_tsc_Y, site_true_Y)
+        self.assertAlmostEqual(py_site_tsc_Y, site_true_Y)
+        self.assertAlmostEqual(site_tsc.tree_stat(A, f), site_true_Y)
+        self.assertAlmostEqual(py_site_tsc.tree_stat(A, f), site_true_Y)
+
+    def test_small_sim(self):
+        orig_ts = msprime.simulate(4, random_seed=self.random_seed,
+                                   mutation_rate=0.0,
+                                   recombination_rate=3.0)
+        ts = tsutil.jukes_cantor(orig_ts, num_sites=3, mu=3,
+                                 multiple_per_node=True, seed=self.seed)
+        branch_tsc = msprime.BranchLengthStatCalculator(ts)
+        py_branch_tsc = PythonBranchLengthStatCalculator(ts)
+        site_tsc = msprime.SiteStatCalculator(ts)
+        py_site_tsc = PythonSiteStatCalculator(ts)
+
+        for t in ts.trees():
+            print(t.interval)
+            print(msprime.drawing.draw_tree(t, format='unicode'))
```

Comment:
This causes an exception on py27

```  
File "/home/travis/build/jeromekelleher/msprime/msprime/drawing.py", line 78, in draw_tree
    raise ValueError("Unicode tree drawing not supported on Python 2")
ValueError: Unicode tree drawing not supported on Python 2
```

but it's nice to have drawing for debug output. maybe enclose in a try/catch?

Reply:
Use ``format='ascii'`` if you want to use py27. I guess we could add another format called 'text' which chooses unicode on py3 and ascii on py27.

Reply:
thx @ashander - I have fixed these (deleting the second one)